folly/IPAddress.cpp

```diff
@@ -246,12 +246,12 @@ IPAddress::IPAddress(const sockaddr* addr) : addr_(), family_(AF_UNSPEC) {
   family_ = addr->sa_family;
   switch (addr->sa_family) {
     case AF_INET: {
-      const sockaddr_in* v4addr = reinterpret_cast<const sockaddr_in*>(addr);
+      const auto v4addr = reinterpret_cast<const sockaddr_in*>(addr);
```

Comment:
This looks like it's not just a use-`auto`-where-appropriate transformation. If `v4addr` were subsequently reassigned to point to a different object, then the original code would compile but the code after clang-tidy transformation would fail to compile. The pedantically-correct transformation would be from `const sockaddr_in*` to `auto`, not to `const auto`. Is this a bug in clang-tidy?

There can certainly be a *separate* code transformation which `const`-qualifies local variables which are not subsequently modified, but the two code transformations seem like they should be independent.

Reply:
You're right. const auto evaluates to const (const sockaddr_in). Will fix.

Reply:
I think const auto evaluates to const (const sockaddr_in*) while const auto* evaluates to const sockaddr_in* . Seems to be a subtlety in C++.